src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/FluidPump.java

```diff
@@ -153,6 +167,24 @@ protected void tick(@Nonnull Block b) {
                         nextFluid.setType(Material.AIR);
                     }
 
+                    return;
+                }
+                if (SlimefunUtils.isItemSimilar(menu.getItemInSlot(slot), emptyBottle, true, false)) {
+                    ItemStack glassBottle = getFilledBottle(fluid);
```

Comment:
bottle, not glass bottle - glass would indicate empty, waterBottle found indicate filled so imo bottle indicates either possible state?